core/src/datayoga_core/blocks/map/tests/test_map.py

```diff
@@ -51,6 +51,26 @@ async def test_map_field_double_nested_expression_jmespath():
     )
 
 
+@pytest.mark.asyncio
+async def test_map_field_nested_expression_jmespath_object():
+    block = Block(properties={"language": "jmespath",
+                              "expression": {"new_field": "`hello`", "name" : "details.fname"}})
+    block.init()
+    assert await block.run([{"details": {"fname": "john", "lname": "doe"}}]) == utils.all_success([
```

Comment:
this failed previously? I think the main thing you fixed was conversion from yaml to json that cause map to choke, no? perhaps we should add that test?

Reply:
Exactly. It didn't work before.

Right. I created own converter from the json comes from yaml to the jmespath expression.

Reply:
so this test should be a yaml? we need a test that simulates whatever failed previously

Reply:
Not actually. Yaml parsing as object anyway as well as json. So, it's exactly the same.